src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -1875,7 +1899,7 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 			this.logService.trace(`Checking updates for extensions`, infos.map(e => e.id).join(', '));
 			const galleryExtensions = await this.galleryService.getExtensions(infos, { targetPlatform, compatible: true, productVersion: this.getProductVersion(), updateCheck: true }, CancellationToken.None);
 			if (galleryExtensions.length) {
-				await this.syncInstalledExtensionsWithGallery(galleryExtensions);
+				await this.syncInstalledExtensionsAndFlagMissingFromGallery(galleryExtensions);
```

Comment:
This is a good trigger for flagging missing gallery extensions. But this will not work in the following scenario:

- When auto check updates are disabled (`"extensions.autoCheckUpdates" = false`) and if user searches for the extension manually. 

Lets think if we can intercept this flow.

Reply:
Added in https://github.com/microsoft/vscode/pull/248852/commits/b216502ec41ad540fe9967941fd248e0c0b099a8

https://github.com/user-attachments/assets/506cc6a5-ec8a-42c7-8a41-13a0a1481b0a